[RFC] Stage 2: Introducing new fields in file/dll/process fields

AsuNa-jp opened the pull request:
<!--
Thank you for your interest in and contributing to ECS! There are a
few simple things to check before submitting your pull request that
can help with the review process. You should delete these items from
our submission, but they are here to help bring them to your attention.
-->

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
Yes
- Have you followed the [contributor guidelines](https://github.com/elastic/ecs/blob/main/CONTRIBUTING.md)?
Yes
- For proposing substantial changes or additions to the schema, have you reviewed the [RFC process](https://github.com/elastic/ecs/blob/main/rfcs/README.md)?
Yes
- If submitting code/script changes, have you verified all tests pass locally using `make test`?
- If submitting schema/fields updates, have you generated new artifacts by running `make` and committed those changes?
Yes
- Is your pull request against main? Unless there is a good reason otherwise, we prefer pull requests against main and will backport as needed.
Yes
- Have you added an entry to the [CHANGELOG.next.md](https://github.com/elastic/ecs/blob/main/CHANGELOG.next.md)?
Yes

mjwolf commented on schemas/dll.yml:
```diff
@@ -45,3 +45,17 @@
+      type: keyword
+      ignore_above: 8192
+      description: The URL of the webpage that linked to the dll file.
+      example: http://example.com/article1.html
```
```suggestion
      example: http://example.com/article1.html
      beta: This field is beta and subject to change.
```

Please add `beta: This field is beta and subject to change.` here and to all the other new fields that are being added.

AsuNa-jp replied:
Thank you for the feedback! I've fixed as suggested.

mjwolf commented on schemas/process.yml:
```diff
@@ -497,3 +497,17 @@
       description: >
         The length of bytes skipped.
+
+    - name: origin_referrer_url
```
For process, I think it might be better to use `executable.origin_referrer_url` and `executable.origin_url`, since this is a property of the executable file that was used to create the process, not of the process itself.

This is something that's being discussed in otel for a different change: https://github.com/open-telemetry/semantic-conventions/pull/1737. ECS doesn't currently follow this convention to refer to the source executable file all the time (for example with `process.name`, but I think the otel convention is better, and something we should try to do here as well.

trisch-me replied:
@mjwolf I agree with @AsuNa-jp regarding restriction we currently have with executable being existing leaf [node](https://www.elastic.co/guide/en/ecs/current/ecs-process.html#field-process-executable). We have discussed that probably we would need to make adjustments in ECS to accommodate Otel schema. But I think it shouldn’t be a problem for this PR unless you have other ideas for naming

mjwolf replied:
I didn't think of executable already being a keyword, and the problems that would introduce when I made this suggestion. I agree it's best to leave this as it is for now

mjwolf commented on rfcs/text/0048-fileorigin-fields.md:
```diff
@@ -82,6 +82,44 @@ Example sources of data is shown in the above.
 Stage 2: Included a real world example source document. Ideally this example comes from the source(s) identified in stage 1. If not, it should replace them. The goal here is to validate the utility of these field changes in the context of a real world example. Format with the source name as a ### header and the example document in a GitHub code block with json formatting, or if on the larger side ...
 -->
 
+The following is the real world example and usage for these fields.
```
If you have an example of how this can be used on macos, I think it would be good to include here too

mjwolf commented:
@AsuNa-jp Can you add a changelog entry in `CHANGELOG.next.md`?

AsuNa-jp commented:
Hi @mjwolf 
Thank you very much for the feedback. I've added changelog entry to `CHANGELOG.next.md` and added `beta: This field is beta and subject to change.` to all of the newly added fields.

I also added an example of how file origin information is added when downloading a file from a web browser on macOS. Let me know if you need additional examples.

For `process.executable` see my reply at https://github.com/elastic/ecs/pull/2441#discussion_r1942195161
